junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -330,12 +380,16 @@ private void resetPermissionsAndTryToDeleteAgain(Path path, IOException exceptio
 		}
 
 		@SuppressWarnings("ResultOfMethodCallIgnored")
-		private static void resetPermissions(Path path) {
-			File file = path.toFile();
-			file.setReadable(true);
-			file.setWritable(true);
-			if (Files.isDirectory(path)) {
-				file.setExecutable(true);
+		private static void tryToResetPermissions(Path path) {
+			try {
+				File file = path.toFile();
+				file.setReadable(true);
+				file.setWritable(true);
+				if (Files.isDirectory(path)) {
+					file.setExecutable(true);
+				}
+			}
+			catch (UnsupportedOperationException ignore) {
```

Comment:
I couldn't find an easy way to test this change.

I tried to create a test with an anonymous subtype of `Path` where `toFile` throws an `UnsupportedOperationException` and all the other methods are delegated to a `Path` created with `Files.createTempDirectory`, but the subtype causes failures before reaching this part of the code where `sun.nio.fs.UnixPath` is expected (I'm on Mac OS).

Any suggestions or should I leave this uncovered?

Another option could be to have one of the in-memory file system libraries as a test dependency and use it to cover this change. Would that make sense?

Reply:
Solved in 24b6a0b with integration tests for both MemoryFileSystem and Jimfs.